src/assets/scripts/client/canvas/CanvasController.js

```diff
@@ -1217,6 +1218,144 @@ export default class CanvasController {
         cc.restore();
     }
 
+    /**
+     * Draw the `MeasureTool` path and text labels
+     *
+     * @for CanvasController
+     * @method _drawMeasureTool
+     * @param cc {HTMLCanvasContext}
+     * @private
+     */
+    _drawMeasureTool(cc) {
+        const { measureTool } = prop.input;
+
+        if (!measureTool.hasLegs) {
+            return;
+        }
+
+        const pathInfo = measureTool.getPathInfo();
+
+        this._drawMeasureToolPath(cc, pathInfo);
+        this._drawMeasureToolLabels(cc, pathInfo);
+
+        cc.restore();
+    }
+
+    /**
+     * Draw the `MeasureTool` text labels
+     *
+     * @for CanvasController
+     * @method _drawMeasureToolLabels
+     * @param cc {HTMLCanvasContext}
+     * @param pathInfo {object} as returned by `MeasureTools.getPointsAndLabels()`
+     * @private
+     */
+    _drawMeasureToolLabels(cc, pathInfo) {
+        let leg = pathInfo.firstLeg.next;
+        const values = [];
+
+        // This way the points are translated only once
+        while (leg != null) {
+            const p = CanvasStageModel.translatePostionModelToPreciseCanvasPosition(leg.midPoint);
```

Comment:
does `p` stand for `position` or `point`?

Reply:
Yeah, they're control positions in on the Canvas. Fixed